2023/08/wtchangdm/day8.rs

```diff
@@ -0,0 +1,190 @@
+use std::collections::HashMap;
+
+#[derive(Debug, Clone, Copy)]
+enum Direction {
+    Left,
+    Right,
+}
+
+impl Direction {
+    fn from(ch: &char) -> Self {
+        match ch {
+            'L' => Self::Left,
+            'R' => Self::Right,
+            _ => panic!("come on it's impossible"),
+        }
+    }
+}
+
+struct Map {
+    cursor: usize,
+    directions: Vec<Direction>,
+    nodes: HashMap<String, [String; 2]>,
+}
+
+impl Map {
+    fn from(input: &[String]) -> Self {
+        let mut nodes: HashMap<String, [String; 2]> = HashMap::new();
+        let mut iter = input.iter().peekable();
```

Comment:
There is no usage of `peek()`.

```suggestion
        let mut iter = input.iter();
```

Reply:
I was trying to use peek() at first and forgot to remove this as well. Thanks for spotting this.